okhttp/src/main/java/io/grpc/okhttp/OkHttpClientTransport.java

```diff
@@ -493,10 +493,19 @@ public void close() {}
           if (proxy == null) {
             sock = new Socket(address.getAddress(), address.getPort());
           } else {
-            sock = createHttpProxySocket(
-                address, proxy.proxyAddress, proxy.username, proxy.password);
+            if (proxy.getProxyAddress() instanceof InetSocketAddress) {
+              sock = createHttpProxySocket(
+                  address,
+                  (InetSocketAddress)proxy.getProxyAddress(),
+                  proxy.getUsername(),
+                  proxy.getPassword()
+              );
+            } else {
+              throw Status.INTERNAL.withDescription(
+                  "Unsupported SocketAddress implementation " + proxy.getProxyAddress().getClass())
+                  .asException();
```

Comment:
asRuntimeException

Reply:
It has to be asException(), because of the catch it is within.